internal/serverinstall/nomad.go

```diff
@@ -1394,11 +1428,11 @@ func (i *NomadInstaller) UpgradeFlags(set *flag.Set) {
 		Usage:  "Nomad host volume name.",
 	})
 
-	set.BoolVar(&flag.BoolVar{
-		Name:    "nomad-consul-service",
-		Target:  &i.config.consulService,
+	set.StringVar(&flag.StringVar{
+		Name:    "nomad-service-provider",
+		Target:  &i.config.serviceProvider,
 		Usage:   "Create service for Waypoint UI and Server in Consul.",
-		Default: false, // default to false, make sure people opt into this for upgrades
+		Default: "consul",
```

Comment:
This could be a good case for an `EnumVar` so we only ever allow `nomad` or `consul` and can fail early if they give us an unsupported option!

Reply:
@briancain in order to still support an install method without a service discovery provider (pre-Nomad 1.3.0), I'll probably name this 3rd option (no service discovery provider) `none`.